packages/react/src/table/TableScroller.tsx

```diff
@@ -0,0 +1,333 @@
+import {
+	type ReactNode,
+	useCallback,
+	useEffect,
+	useRef,
+	useState,
+} from 'react';
+import { Box } from '../box';
+import { boxPalette } from '../core';
+import { Flex } from '../flex';
+import { ScrollbarArrowLeftIcon, ScrollbarArrowRightIcon } from '../icon';
+import { Stack } from '../stack';
+
+export type TableScrollerProps = { children: ReactNode };
+
+export function TableScroller({ children }: TableScrollerProps) {
+	const trackRef = useRef<HTMLDivElement>(null);
+	const thumbRef = useRef<HTMLButtonElement>(null);
+	const scrollerRef = useRef<HTMLDivElement>(null);
+	const mousePos = useRef({ x: 0, y: 0 });
+
+	const [isDraggingThumb, setIsDraggingThumb] = useState(false);
+	const [thumbPosition, setThumbPosition] = useState(0);
+	const [thumbWidthRatio, setThumbWidthRatio] = useState(0);
+	const [buttonIntervalId, setButtonIntervalId] = useState<number | null>(null);
+
+	const repositionThumb = useCallback(() => {
+		if (!scrollerRef?.current) {
+			return;
+		}
+
+		setThumbPosition(scrollerRef.current.scrollLeft * thumbWidthRatio);
+	}, [scrollerRef, thumbWidthRatio]);
+
+	const calculateThumbWidth = useCallback(() => {
+		if (!scrollerRef?.current) {
+			return;
+		}
+
+		setThumbWidthRatio(
+			(scrollerRef?.current.offsetWidth - 56) /
+				(scrollerRef?.current.scrollWidth - 56)
+		);
+	}, [scrollerRef, setThumbWidthRatio]);
```

Comment:
Same here re: removing `scrollerRef`. Also setter functions returned from `useState` are guaranteed to be stable across re-renders so you don't need `setThumbWidthRatio` either. This dependency array can just be empty.